app/Filament/Training/Pages/ExamSetup.php

```diff
@@ -0,0 +1,271 @@
+        $position = CtsPosition::find($positionId);
+
+        $ctsMember = Member::where('cid', $this->dataOBS['student_obs'])->first();
+
+        $setup = ExamSetupModel::create([
+            'rts_id' => 14, // hard coded for OBS
+            'student_id' => $ctsMember->id,
+            'position_1' => $position->callsign,
+            'position_2' => null,
+            'exam' => 'OBS',
+        ]);
+
+        $examBooking = ExamBooking::create([
+            'rts_id' => 14, // hard coded for OBS
+            'student_id' => $ctsMember->id,
+            'student_rating' => $ctsMember->account->qualification_atc->vatsim,
+            'position_1' => $position->callsign,
+            'position_2' => null,
+            'exam' => 'OBS',
+        ]);
+
+        $setup->update([
+            'bookid' => $examBooking->id,
+        ]);
+
+        Notification::make()
+            ->title('Exam Setup')
+            ->success()
+            ->body('Exam setup for '.$position->callsign.' has been created.')
+            ->send();
+
+        return redirect()->route('filament.training.pages.exam-setup');
+    }
+
+    public function formOBS(Form $form): Form
+    {
+        return $form
+            ->schema([
+                Section::make('Exam Setup - OBS PT3')
+                    ->schema([
+                        Select::make('position_obs')
+                            ->label('Position')
+                            ->options(CtsPosition::where('callsign', 'LIKE', 'OBS_%_PT3')->orderBy('callsign')->pluck('callsign', 'id'))
+                            ->required()
+                            ->searchable()
+                            ->preload()
+                            ->live()
+                            ->afterStateUpdated(function ($state, callable $set) {
+                                $set('student_obs', null);
+                            }),
+
+                        Select::make('student_obs')
+                            ->label('Student')
+                            ->options(function (Get $get): array {
+                                // show recent students who have taken a session at the position
+                                // PT2 i.e. OBS_XX_PT2. The position selected in the dropdown is
+                                // going to be PT3. So we need to get the PT2 position.
+                                // XX is the last two characters of the position callsign first part.
+
+                                $positionId = $get('position_obs');
+                                if ($positionId == null) {
+                                    return [];
+                                }
+
+                                $position = CtsPosition::find($positionId);
+                                $pt2Position = CtsPosition::where('callsign', 'LIKE', Str::replaceLast('PT3', 'PT2', $position->callsign))->first();
+
+                                $recentPassedStudentIds = (new ExamResultRepository)
```

Comment:
why do recently passed students appear here for setting up exams?

Reply:
They are used to exclude rather than include